### PR TITLE
Refactor FXIOS-3619 [v106] Tab manager store changes to make it testable + adding tests

### PR DESCRIPTION
**Pull request by lmarceau**
# [FXIOS-3619](https://mozilla-hub.atlassian.net/browse/FXIOS-3619) https://github.com/mozilla-mobile/firefox-ios/issues/9770
Wanted to add tests before looking into fixing deprecations issues in the TabManagerStore.
- Separate SiteArchiver so it returns normal tabs. SiteArchiver was a static struct so it could be called in widget to get widget tabs (named SimpleTabs). Widget tabs are retrieved directly through SimpleTab. Some logic from SiteArchiver was moved to TabManagerStore. Some was moved in TabDataRetriever.
- Add new TabFileManager protocol to have more control over tests data.
- Remove writeCompletion as now tests are sync, since we use a mock serialQueue in the tests.
- restoreStartupTabs function passes a closure `addTabClosure` instead of passing the TabManager object to the TabManagerStore. In the case of a nil tabToSelect, the logic was moved into TabManager itself as it is the TabManager that selects tabs.

### CODE REVIEW

**Review by OrlaM — APPROVED** [review]
Small comment about a unused import but otherwise this looks great. 
I'd recommend getting a second review on this tho, just because it's a high risk part of the code.

**Comment by lmarceau** [comment]
Build is green [here](https://app.bitrise.io/app/6c06d3a40422d10f/pipelines/d25c1096-7fa0-479c-a0a4-a5a02a413ea8)
Screen Shot 2022-09-08 at 12 08 32 PM

File: Client/Frontend/Browser/Tab Management/TabManagerStore.swift
Comment (nbhasin2) [inline Client/Frontend/Browser/Tab Management/TabManagerStore.swift:152]: Wondering if we should log this in sentry as well 🤔

Comment (lmarceau) [reply Client/Frontend/Browser/Tab Management/TabManagerStore.swift:152]: Good point, I can make the change in my subsequent PR where I want to fix the deprecation warning

**Review by nbhasin2 — APPROVED** [review]

**Pull request merged**
